### PR TITLE
Ko/add auth for icarus pretty

### PR DESCRIPTION
**Pull request by ibihim**

### CODE REVIEW

File: src/authCloud/index.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+const config = ({ clientId, clientURL, gcSDK }) => {
+    const postKeys = ({ privateKey, publicKey }) =>
+        request.post(`${clientURL}/gckeys`)
+            .send({ private_key: privateKey, public_key: publicKey })
+            .then(identity);
+
+    const getToken = () =>
+        request.get(`${clientURL}/gctoken`).then(({ text }) => {
+            const resObj = JSON.parse(text) || {};
```
Comment (Glumli) [inline src/authCloud/index.js]: Did you check if there is a body?

File: example/server.js [hunk 2]
```diff
@@ -1,10 +1,20 @@
...
-let port = 8888;
+const express = require('express');
+const cors = require('cors');
+const proxy = require('http-proxy-middleware');
+const app = express();
+const host = '0.0.0.0';
+const port = 8888;
 
 app.use('/', express.static('./'));
+app.use('/icarus', proxy({
```
Comment (theintz) [inline example/server.js]: I would suggest not to use the cryptic (for outsiders) names as routes.

File: src/authCloud/index.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                onceElse(
+                    () => Promise.resolve(token),
+                    getToken,
+                ),
+            );
+
+            return true;
+        }
+
+        cap.then(postKeys).then(console.log);
```
Comment (Glumli) [inline src/authCloud/index.js]: you wanted to rename this

Comment (Glumli) [reply src/authCloud/index.js]: console.log

File: example/server.js [hunk 4]
```diff
@@ -1,10 +1,30 @@
...
+const https = require('https');
+const cors = require('cors');
+const proxy = require('http-proxy-middleware');
+const host = '0.0.0.0';
+const port = 8888;
 
-let app = express();
-let host = '0.0.0.0';
-let port = 8888;
+const privateKey = fs.readFileSync('sslcert/server.key', 'utf8');
```
Comment (Glumli) [inline example/server.js]: Please don't make the keys required. Let's check if we really need those for local development

File: src/authCloud/index.js (hunk 1 above)
Comment (Glumli) [reply src/authCloud/index.js]: ok fail

File: src/authCloud/index.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+            sdk.setup(
+                clientId,
+                privateKey,
+                onceElse(
+                    () => Promise.resolve(token),
+                    getToken,
+                ),
+            );
+
+            return true;
```
Comment (Glumli) [inline src/authCloud/index.js]: return true after setup, but not in parallel.

**Pull request closed without merging**